sweepbatcher/sweep_batch.go

```diff
@@ -920,6 +943,25 @@ func (b *batch) Run(ctx context.Context) error {
 				skipBefore = &delayStop
 				skipBeforeUpdated = true
 			}
+
+			// Check if batch exceeds threshold and should publish
+			// immediately. Only override skipBefore if threshold
+			// feature is enabled and exceeded.
+			if b.exceedsThreshold() {
+				b.Infof("Batch value (%v) exceeds threshold "+
+					"(%v), publishing immediately",
+					b.totalValue(),
+					b.cfg.immediatePublishThreshold)
+
+				// Set skipBefore to now to bypass initial
+				// delay.
+				now := clock.Now()
+				skipBefore = &now
+
+				// Trigger immediate publish by setting timer to
+				// fire after batchPublishDelay.
+				timerChan = clock.TickAfter(b.cfg.batchPublishDelay)
```

Comment:
Simplification idea.

I propose to move the whole branch above, under `if initialDelay < 0 {`
There we already have a couple of situations where we skip waiting. For consistency it is better to have them all in one place. And skipping can also be done by setting `initialDelay = 0`.

I propose also to add an empty line before `delayStop := startTime.Add(initialDelay)`